### ISSUE TITLE
Unable to align commit detail DOM elements with vertical reverse orientation

### ISSUE THREAD

**Issue by csombok** [nicoespeon/gitgraph.js#167]
When using vertical-reverse orientation commit points are always aligned to the bottom of the detail DOM element and there are no options to customize it (i.e. vertical middle, or top)

Example:

![image](https://cloud.githubusercontent.com/assets/12937212/26832486/ec47ba86-4acf-11e7-9702-6518b7fe13d7.png)

Example code:

      var gitgraph = new GitGraph({
        template: "blackarrow",
        orientation: "vertical-reverse",
      });

      gitgraph.template.commit.message.display = false;

      var project = gitgraph.branch("project");
      project.commit({ message: "detail-1", detailId: "detail-1" })
      var feature = project.branch("feature");
      feature.commit({ message: "detail-2", detailId: "detail-2" })
      feature.merge(project, { message: "detail-3", detailId: "detail-3" });
    }

**Comment by nicoespeon** [comment]
Hi @csombok,

I see what you mean. Indeed, there is no possibility to change the way details are rendered. This should be an option − or I least, I'd find more intuitive for details to be aligned by the top rather than the bottom.

**Comment by ch4nn0n** [comment]
I am having this issue as well, it would be nice to have the details aligned under the commit points.

**Comment by nicoespeon** [comment]
A little update on this one: I started fixing this, then I realize it won't be solved easily because of the way graph rendering is currently implemented.

For now, when the `.commit()` function is called, the commit is rendered and a shared state is set so the next commit will have a margin so it doesn't collide with the detail.
However, to render the detail as the top in `vertical-reverse` mode, we'd need the *previous commit* to be spaced so it doesn't collide. Hence, this is not easily achievable because of the current processing order of rendering + the use of a shared state to do that (= can't easily "correct" commits that have been rendered before).

Then, I fear this is not something we could do until a v2 of the lib − which is something poking around my mind, but I don't have time / motivation for that so I can't promise a fix anytime soon.

**Comment by nicoespeon** [comment]
So, this is technically _not_ resolved for the `gitgraph.js` package… But we released `@gitgraph/react` in v1.0.0-0 and this issue was fixed here. We even have a story for that:

![image](https://user-images.githubusercontent.com/1094774/52529147-e213d300-2cba-11e9-8a5e-316eb469767b.png)

```jsx
const renderMessage: CommitOptions["renderMessage"] = (
  commit,
  commitMessageX,
) => {
  return (
    <g
      transform={`translate(${commitMessageX - commit.x}, ${
        commit.style.dot.size
      })`}
    >
      <text fill={commit.style.dot.color}>
        {commit.hashAbbrev} - {commit.subject}
      </text>
      <foreignObject width="600" x="10">
        <p>
          My money's in that office, right? If she start giving me some
          bullshit about it ain't there, and we got to go someplace else and
          get it, I'm gonna shoot you in the head then and there. Then I'm
          gonna shoot that bitch in the kneecaps, find out where my goddamn
          money is. She gonna tell me too. Hey, look at me when I'm talking
          to you, motherfucker. You listen: we go in there, and that nigga
          Winston or anybody else is in there, you the first motherfucker to
          get shot. You understand?
        </p>
      </foreignObject>
    </g>
  );
};

return (
  <Gitgraph options={{ orientation: Orientation.VerticalReverse }}>
    {(gitgraph) => {
      gitgraph
        .commit({ subject: "Initial commit" })
        .commit({ subject: "Another commit" })
        .commit({
          subject: "Do something crazy",
          renderMessage,
        });

      gitgraph
        .branch("dev")
        .commit({
          subject: "Oh my god",
        })
        .commit({
          subject: "This is a saxo!",
        });
    }}
  </Gitgraph>
);
```

Thus, I'm closing this issue as we won't maintain the `gitgraph.js` package. There will be a `@gitgraph/js` package soon to replace it. 

We still accept PRs if someone really needs it, but I'd suggest moving to the aforementioned packages. Thanks 🙏
